File: Python/leetcode_905.py
from typing import List


def sort_array_by_parity(nums: List[int]) -> List[int]:
    result: list[int] = []
    for num in nums:
        if num % 2 == 0:
            result.insert(0, num)
        else:
            result.append(num)
    return result

File: JavaScript/leetcode_905.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
    let j = 0;
    for (let index = 0; index < nums.length; index++) {
        if (nums[index] % 2 === 0) {
            [nums[index], nums[j]] = [nums[j], nums[index]];
            j++;
        }
    }
    return nums;
};
